core/repos/repomanager.go

```diff
@@ -0,0 +1,389 @@
+/*
+ * === This file is part of ALICE O² ===
+ *
+ * Copyright 2019 CERN and copyright holders of ALICE O².
+ * Author: Kostas Alexopoulos <kostas.alexopoulos@cern.ch>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In applying this license CERN does not waive the privileges and
+ * immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package repos
+
+import (
+	"errors"
+	"github.com/spf13/viper"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport/http"
+	"io"
+	"log"
+	"os"
+	"sort"
+	"strings"
+	"sync"
+)
+
+var (
+	once sync.Once
+	instance *RepoManager
+	gitAuthUser = "kalexopo"
```

Comment:
Is a single auth mechanism/credentials used for all the repos? I figure it should be per-repo.
I also expect it shouldn't require additional configuration storage, as once the repo is cloned the credentials should be stored in `.git/config` (except with password auth, but we can choose not to support this).